packages/block-library/src/template-part/variations.js

```diff
@@ -6,55 +6,57 @@ import { store as editorStore } from '@wordpress/editor';
 import { store as blocksStore } from '@wordpress/blocks';
 import { dispatch, select, subscribe } from '@wordpress/data';
 
-const unsubscribe = subscribe( () => {
-	const definedVariations = select(
-		editorStore
-	).__experimentalGetDefaultTemplatePartAreas();
+export default function subscribeToCreateVatiations() {
+	const unsubscribe = subscribe( () => {
```

Comment:
The global subscribe/select/dispatch is something we avoid using as much as we can. For what reason are we subscribing here? I assume we're waiting for something but what is this thing we're waiting for here?

(We should probably document this somewhere)

Reply:
Its waiting for the template part area definitions from the php side to initialize in the editor store.  

Im looking at refactoring to avoid subscribe and register these variations from the server side here -https://github.com/WordPress/gutenberg/pull/31761

Reply:
It looks like that fix is hit or miss for some folks, still have to investigate its inconsistencies.  

For now, I would think this `subscribe` may be alright?  Technically its only present until the store initializes.

Reply:
> It looks like that fix is hit or miss for some folks, still have to investigate its inconsistencies.

Nm, that seems to be a WP version issue that we should be able to work around with fallbacks.